resources/js/validation.js

```diff
@@ -0,0 +1,188 @@
+import { ErrorMessage } from './validation_message.js';
+import { initPasswordToggles } from './ui_common.js';
+
+document.addEventListener('DOMContentLoaded', () => {  
+  initPasswordToggles(); // 비밀번호 표시/숨김 아이콘 초기화
+
+  const loginForm = document.getElementById('myFormlogin');
+  const signupForm = document.getElementById('myFormSignUp');
+  const emailInput = document.getElementById('email');
+  const passwordInput = document.getElementById('password');
+
+  // 폼과 제출 버튼, 페이지 유형을 저장할 변수들
+  let form, submitButton, nicknameInput, confirmPasswordInput;
+  let isLoginPage = false;
+  let isSignupPage = false;
+
+  // 폼 존재 여부를 확인 후 알맞은 값을 할당
+  if (loginForm !== null) {
+    isLoginPage = true;
+    form = loginForm;
+    submitButton = form.querySelector('.login');
+  } else if (signupForm !== null) {
+    isSignupPage = true;
+    form = signupForm;
+    submitButton = form.querySelector('.singup');
+  } else {
+    return false;
+  }
+  
+  // 회원가입 페이지에서만 필요한 요소들 
+  if (isSignupPage) {
+    nicknameInput = document.getElementById('nickname');
+    confirmPasswordInput = document.getElementById('confirmPassword');
+  }
+  
+  // 에러 메시지 표시 함수
+  function showError(input, message) {
+    input.style.borderColor = '#F74747';
+    const newErrorElement = document.createElement('p');
+    newErrorElement.className = 'error-message';
+    newErrorElement.textContent = message;
+    newErrorElement.style.color = '#F74747';
+    newErrorElement.style.fontWeight = '600';
+    newErrorElement.style.margin = '10px 15px 0';
+    newErrorElement.style.display = 'block';
+    input.parentElement.appendChild(newErrorElement);
+  }
```

Comment:
💊 제안
해당 함수에서 input의 borderColor 및 오류 메시지 스타일을 js로 직접 설정하게 해주셨네요~
그러나 스타일링은 css에서 관리하는 것이 유지보수에 더 좋을 것 같습니다.
해당 함수에서는 클래스만 추가하고, 스타일은 css에서 적용하는 방식으로 변경하시는 것을 추천드려요!

```suggestion
  function showError(input, message) {
    input.style.borderColor = '#F74747';
    
    const newErrorElement = document.createElement('p');
    
    newErrorElement.className = 'error-message';
    newErrorElement.textContent = message;
    
    input.parentElement.appendChild(newErrorElement);
  }
```